Riot/AppDelegate.m

```diff
@@ -1097,6 +1113,46 @@ - (void)application:(UIApplication *)application didRegisterUserNotificationSett
     }
 }
 
+- (void)application:(UIApplication *)application handleActionWithIdentifier:(NSString *)identifier forLocalNotification:(UILocalNotification *)notification withResponseInfo:(NSDictionary *)responseInfo completionHandler:(void (^)())completionHandler
+{
+    if ([identifier isEqualToString: @"inline-reply"])
+    {
+        NSString* roomId = notification.userInfo[@"room_id"];
+        if (roomId.length)
+        {
+            NSArray* mxAccounts = [MXKAccountManager sharedManager].activeAccounts;
+            MXRoom* room = nil;
+            for (MXKAccount* account in mxAccounts)
+            {
+                room = [account.mxSession roomWithRoomId:roomId];
+                if (room)
+                {
+                    break;
+                }
+            }
+            if (room == nil)
+            {
+                NSLog(@"[AppDelegate][Push] handleActionWithIdentifier: room with id %@ not found", roomId);
+            }
+            else
+            {
+                NSString* responseText = [responseInfo objectForKey:UIUserNotificationActionResponseTypedTextKey];
+                if (responseText != nil && responseText.length != 0)
+                {
+                    NSLog(@"[AppDelegate][Push] handleActionWithIdentifier: sending message to room: %@", roomId);
+                    [room sendTextMessage:responseText success:^(NSString* eventId) {} failure:^(NSError* error) {
```

Comment:
Can you use the `sendTextMessage` method of `MXKRoomDataSource`, instead?
It does a bit more processing on text before sending. Plus, it will manage local echo if the user opens the room while the message is being sent.